components/script/dom/idbkeyrange.rs

```diff
@@ -111,6 +111,9 @@ impl IDBKeyRangeMethods<crate::DomTypeHolder> for IDBKeyRange {
     ) -> Fallible<DomRoot<IDBKeyRange>> {
         let lower_key = convert_value_to_key(cx, lower, None)?;
         let upper_key = convert_value_to_key(cx, upper, None)?;
+        if lower_key > upper_key {
+            return Err(Error::Data);
+        }
```

Comment:
Nice and simple.

Would you mind adding spec comments to this function, like this:

> // Step 5. If lowerKey is [greater than](https://www.w3.org/TR/IndexedDB-2/#greater-than) upperKey, [throw](https://www.w3.org/TR/WebIDL-1/#dfn-throw) a "[DataError](https://www.w3.org/TR/WebIDL-1/#dataerror)" [DOMException](https://www.w3.org/TR/WebIDL-1/#idl-DOMException).

Having these earlier would have made the bug obvious.